core/src/tile/labels/labels.cpp

```diff
@@ -32,7 +36,13 @@ bool Labels::addLabel(MapTile& _tile, const std::string& _styleName, Label::Tran
             return false;
         }
 
-        l->update(m_view->getViewProjectionMatrix() * _tile.getModelMatrix(), m_screenSize, 0);
+        // NB: viewOrigin.z is only determined by screen width and height.
+        const auto& viewOrigin = m_view->getPosition();
+
+        auto modelMatrix = glm::scale(glm::mat4(1.0), glm::vec3(_tile.getScale()));
+        modelMatrix[3][2] = -viewOrigin.z;
+
+        l->update(m_view->getViewProjectionMatrix() * modelMatrix, m_screenSize, 0);
```

Comment:
Yes, we need the projection matrix to project labels to screen ensuring the label is 'valid' position/state speaking after being created. We could probably store the current projection matrix and position in a member though.